language/e2e-tests/src/tests/transaction_fuzzer.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{account::Account, executor::FakeExecutor};
+use proptest::{collection::vec, prelude::*};
+use transaction_builder_generated::stdlib::ScriptCall;
+
+proptest! {
+    #![proptest_config(ProptestConfig::with_cases(32))]
+    #[test]
+    fn fuzz_scripts(
+        txns in vec(any::<ScriptCall>(), 0..10),
+    ) {
+        let executor = FakeExecutor::from_genesis_file();
+        let mut accounts = vec![];
+        accounts.push((Account::new_libra_root(), 1));
+        accounts.push((Account::new_blessed_tc(), 0));
+        let num_accounts = accounts.len();
+
+        for (i, txn) in txns.into_iter().enumerate() {
+            let script = txn.encode();
+            let (account, account_sequence_number) = &accounts[i % num_accounts];
+            let output = executor.execute_transaction(
```

Comment:
Should we apply the `output.write_set` (or perhaps do this in a different proptest)? I think as it stands, the test is executing a bunch of random txes against the genesis state, which is very useful. But we will see even more behaviors if we can execute txes against non-genesis states as well.

Reply:
I think that's a really good idea! I'll add it as another proptest :)